shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1180,8 +1180,30 @@ - (void)keyboardWillChangeFrame:(NSNotification*)notification {
 }
 
 - (void)keyboardWillBeHidden:(NSNotification*)notification {
-  // When keyboard hide, the keyboardWillChangeFrame function will be called to update viewport
-  // metrics. So do not call [self updateViewportMetrics] here again.
+  NSDictionary* info = [notification userInfo];
+
+  if (@available(iOS 9, *)) {
+    // Ignore keyboard notifications related to other apps.
+    id isLocal = info[UIKeyboardIsLocalUserInfoKey];
+    if (isLocal && ![isLocal boolValue]) {
+      return;
+    }
+  }
+
+  // Ignore keyboard notifications if engine’s viewController is not current viewController.
+  if ([_engine.get() viewController] != self) {
+    return;
+  }
```

Comment:
nits: This code is the same as in keyboardWillChangeFrame, maybe we can refactor it into a method.